### ISSUE TITLE
allow consumer to set promise implementation

### ISSUE THREAD

**Issue by pwmckenna** [yahoo/fluxible#482]
I would love to be able to set the promise library for fluxible, so that my actions return Q promises for instance. There are a bunch of libraries that do this, and the approaches are all pretty straight forward (either a setter, or a exported variable that can be overridden).

Here's a related discussion that happened with mongoose: https://github.com/Automattic/mongoose/issues/2688

Is there interest in supporting something like this? Would be happy to take a crack at a pr.

**Comment by redonkulus** [comment]
@pwmckenna Fluxible's code assumes that a `Promise` implementation is provided when bundled (either using built-in es6 promise or something else). When bundling you could try setting the `Promise` implementation to whatever library you want to use.

**Comment by pwmckenna** [comment]
I can imagine using some webpack magic to use a library of my choosing on the frontend, but how would this work server side?

**Comment by redonkulus** [comment]
In the past, on the server we did something like this global.Promise = require('bluebird'); at the top of your server entry point file.

**Comment by pwmckenna** [comment]
I tried the node change you suggested and it worked fine, but I'm using webpack with the babel-loader, and its unclear whether its possible to make this work in that case. iirc Promise is either provided by js-code or the existing global Promise is used if spec compliant. It doesn't appear that there's a good way to force it to use a specific implementation.

The babel issues that I found also made the point that overriding the global Promise implementation will affect every other library and could have unintended side effects. I understand not wanting to handle this within fluxible, so feel free to just close this if we're at the end of the road :)
